core/templates/dev/head/player/LearnerLocalNav.js

```diff
@@ -83,5 +87,63 @@ oppia.controller('LearnerLocalNav', [
         });
       });
     };
+
+    $scope.showFlagExplorationModal = function() {
+      $modal.open({
+        templateUrl: 'modals/flagExploration',
+        backdrop: true,
+        controller: [
+            '$scope', '$modalInstance',
+          function($scope, $modalInstance) {
+            $scope.otherFlagDropdown = false;
+
+            $scope.triggerOtherFlagDropdown = function(value) {
+              if (value === 'other') {
+                $scope.otherFlagDropdown = true;
+                $scope.flagMessage = null;
+              } else {
+                $scope.otherFlagDropdown = false;
+                $scope.flagMessage = $scope.flag;
+              }
+            };
+
+            $scope.submitReport = function() {
+              if ($scope.flagMessage) {
+                $modalInstance.close({
+                  report: $scope.flagMessage
```

Comment:
In the email sent, can you also include the current state name? This is going to be useful because the moderators will probably need to know where the user was when they filed the report.

You could add it in brackets before the report itself, e.g.

```
   "[name of state] (reason for report) text of report"
```